id-repository/id-repository-identity-service/src/main/java/io/mosip/idrepository/identity/service/impl/IdRepoProxyServiceImpl.java

```diff
@@ -562,12 +562,12 @@ private void getBiometricFiles(Uin uinObject, List<DocumentsDTO> documents, Map<
 			if (allowedBioAttributes.contains(bio.getBiometricFileType())) {
 				try {
 					String uinHash = uinObject.getUinHash().split("_")[1];
-					if (!objectStoreHelper.biometricObjectExists(uinHash, bio.getBioFileId())) {
+					byte[] data = objectStoreHelper.getBiometricObject(uinHash, bio.getBioFileId());
+					if (Objects.isNull(data)) {
 						mosipLogger.error(IdRepoSecurityManager.getUser(), ID_REPO_SERVICE_IMPL, "getBiometricFiles",
 								"FILE NOT FOUND IN OBJECT STORE");
 						throw new IdRepoAppUncheckedException(FILE_NOT_FOUND);
 					}
-					byte[] data = objectStoreHelper.getBiometricObject(uinHash, bio.getBioFileId());
 					if (Objects.nonNull(data)) {

```

Comment:
Please remove this condition as it is already handled in previous condition.
if (Objects.nonNull(data)) {

Reply:
Done